pymobiledevice3/remote/core_device_tunnel_service.py

```diff
@@ -0,0 +1,472 @@
+from cryptography.hazmat.primitives import hashes
+from cryptography.hazmat.primitives._serialization import Encoding, PublicFormat
+from cryptography.hazmat.primitives.asymmetric.ed25519 import Ed25519PrivateKey
+from cryptography.hazmat.primitives.asymmetric.rsa import RSAPrivateKey
+from cryptography.hazmat.primitives.asymmetric.x25519 import X25519PrivateKey, X25519PublicKey
+from cryptography.hazmat.primitives.ciphers.aead import ChaCha20Poly1305
+from cryptography.hazmat.primitives.kdf.hkdf import HKDF
+from srptools import SRPClientSession, SRPContext
+from srptools.constants import PRIME_3072, PRIME_3072_GEN
+
+from pymobiledevice3.ca import make_cert
+from pymobiledevice3.pair_records import create_pairing_records_cache_folder, generate_host_id
+from pymobiledevice3.remote.remote_pairing_service import RemotePairingService
+from pymobiledevice3.remote.remote_service_discovery import RemoteServiceDiscoveryService
+from pymobiledevice3.remote.remotexpc import XpcInt64Type, XpcUInt64Type
+
+PairingDataComponentType = Enum(Int8ul,
+                                METHOD=0x00,
+                                IDENTIFIER=0x01,
+                                SALT=0x02,
+                                PUBLIC_KEY=0x03,
+                                PROOF=0x04,
+                                ENCRYPTED_DATA=0x05,
+                                STATE=0x06,
+                                ERROR=0x07,
+                                RETRY_DELAY=0x08,
+                                CERTIFICATE=0x09,
+                                SIGNATURE=0x0a,
+                                PERMISSIONS=0x0b,
+                                FRAGMENT_DATA=0x0c,
+                                FRAGMENT_LAST=0x0d,
+                                SESSION_ID=0x0e,
+                                TTL=0x0f,
+                                EXTRA_DATA=0x10,
+                                INFO=0x11,
+                                ACL=0x12,
+                                FLAGS=0x13,
+                                VALIDATION_DATA=0x14,
+                                MFI_AUTH_TOKEN=0x15,
+                                MFI_PRODUCT_TYPE=0x16,
+                                SERIAL_NUMBER=0x17,
+                                MFI_AUTH_TOKEN_UUID=0x18,
+                                APP_FLAGS=0x19,
+                                OWNERSHIP_PROOF=0x1a,
+                                SETUP_CODE_TYPE=0x1b,
+                                PRODUCTION_DATA=0x1c,
+                                APP_INFO=0x1d,
+                                SEPARATOR=0xff)
+
+TLV = Struct(
+    'type' / PairingDataComponentType,
+    'data' / Prefixed(Int8ul, GreedyBytes),
+)
+
+TLVBuf = GreedyRange(TLV)
+
+PairConsentResult = namedtuple('PairConsentResult', 'public_key salt')
+
+CDTunnelPacket = Struct(
+    'magic' / Const(b'CDTunnel'),
+    'body' / Prefixed(Int16ub, GreedyBytes),
+)
+
+logger = logging.getLogger(__name__)
```

Comment:
Maybe move to RemotePairingService as a member?

Reply:
done